Update roadmap

nerdneha opened the pull request:
[Rendered Markdown](https://github.com/desktop/desktop/blob/long-and-winding-roadmap/docs/process/roadmap.md)

includes releases and very rough sketch of features for 1.2+
* includes rough idea of releases
* will change based on reality
* we will re-evaluate at the 6 month mark in August
* subject to change as new features are released on GitHub.com or as we discover better opportunities to impact the user positively

![way of the future](https://media.giphy.com/media/eeWuioiY6DnLq/giphy.gif)

donokuda commented on docs/process/roadmap.md:
```diff
@@ -1,63 +1,70 @@
+## 1.3 - Detecting marge conflicts - 2.0
+  
+- [Relationship between branches 2.0](https://github.com/desktop/desktop/issues/2639)
+  - Notifications around diverging from master branch
```
I think @niik's [comment here](https://github.com/desktop/desktop/issues/2639#issuecomment-340791736) explains the notifications part a bit. Should we consider linking this bullet point to there?

donokuda commented on docs/process/roadmap.md:
```diff
@@ -1,63 +1,70 @@
-  - Differentiate conflicted files from other changed files
-  - Let me choose my mergetool
-  - Let me open my mergetool
+## 1.3 - Detecting marge conflicts - 2.0
```
> marge conflicts

![image](https://user-images.githubusercontent.com/1174461/36824086-ad40510c-1ca4-11e8-9f0e-e135fec4040c.png)

shiftkey replied:
Can we just leave this as-is? :trollface:

shiftkey replied:
But more seriously, what's the `2.0` part about this represent? Conflict resolution **lite** is set down for `1.6` - should this be `1.0` instead?

donokuda commented on docs/process/roadmap.md:
```diff
@@ -1,63 +1,70 @@
 
-## 1.2
+- [Conflict resolution lite](https://github.com/desktop/desktop/issues/2640)
+  - Differentiate conflicted files from other changed files
```
"Differentiate" seems to be a little confusing based on some of the comments in that issue. Maybe something like "Separate conflicted files from other changed files"?

donokuda commented on docs/process/roadmap.md:
```diff
@@ -1,63 +1,70 @@
-  - Only on my PRs
-  - Use OS notifications
-  - Notifications link to the PR on dotcom
+- Human readable Table of Contents
```
Maybe something like "View changes by methods and functions (for supported files)?"

I'm assuming we were talking implementing something like [this](https://blog.github.com/2017-07-26-quickly-review-changed-methods-and-functions-in-your-pull-requests/)?

donokuda commented on docs/process/roadmap.md:
```diff
@@ -1,63 +1,70 @@
-- Protected branches and default branch protection
-  - Don't let me commit to branches I shouldn't commit to
-  - Guide me to creating my own branches
+- App interaction tracking
```
> End of progress & in-app messaging
>
> App interaction tracking

I.. totally forgot what these were. Can any shed some clarity for me? 😅

shiftkey replied:
@donokuda attempting to remember our earlier discussions:

 - "End of Progress" was calling out some of the flows where we don't make it clear that they're done - we just wrap it up and return the user to the main view. The classic example of this is login - it's not clear that your login succeeded, the flow just finishes.

 - I'm a bit hazy on "in-app messaging" and can't remember the concrete things we talked about

 - "App interaction tracking" - as we roll out features, it'd be great to understand just how much they're being used in the wild. We've been haphazard about this in the past, so this was a chance for us to revisit this.

donokuda replied:
Crystal clear now 💎thanks, @shiftkey!

shiftkey commented on docs/process/roadmap.md:
```diff
@@ -1,63 +1,71 @@
+- Human readable Table of Contents
 
-## 1.3
+## Icebox
```
This is totally a leading question, and overlaps with what I've been pondering about, but does this stuff belong on the roadmap? Or can it live elsewhere?

shiftkey commented on docs/process/roadmap.md:
```diff
@@ -1,63 +1,71 @@
-  - Show how the current branch relates to the target branch
-  - The target is the default branch or the upstream's default branch
-  - _Probably_ let the user switch the target branch
+- [Relationship between branches lite](https://github.com/desktop/desktop/issues/3956) - [#3956](https://github.com/desktop/desktop/issues/3956)
```
We don't explain what "lite", "1.0" and "2.0" represent in terms of this document. I took a stab at this here, see if you like it or not:

> A feature may be suffixed with a label like "lite" or "1.0" or "2.0" - this is used to define the scope of the feature, as it is expected to evolve over time:
>
> - "lite" - the fundamentals of the feature are implemented, enough to get the common tasks done
> - "1.0" - the feature has evolved beyond the fundamentals, enough for every day use
> - "2.0" - with more knowledge about the domain and use cases, the feature now supports advanced tasks

nerdneha commented:
@shiftkey @donokuda 

Not sure if this was too 🔥 ✊ but I pushed up a version without `lite` and `2.0` to see if it was possible. Instead, I used `MVP` and `continued`. Is this still confusing? Is it worse? Would love to hear your thoughts. 

also thanks for the rest of the feedback! removed the icebox and added some clarity around later releases. more than happy to go in and add even more detail!

nerdneha commented:
🦑

niik commented:
Gonna give this to @donokuda to merge just to verify that he's happy with the changes.

niik commented:
@nerdneha I'm assuming this was `ready-for-review`?

shiftkey commented:
>  instead, I used MVP and continued. is this still confusing? Is it worse? would love to hear your thoughts.

@nerdneha 👍 yeah, this works

donokuda commented:
![giphy 4](https://user-images.githubusercontent.com/1174461/36991647-34ea1e9c-205d-11e8-9747-3a1c8195ba15.gif)
